mat64/matrix.go

```diff
@@ -883,3 +883,12 @@ func zero(f []float64) {
 		f[i] = 0
 	}
 }
+
+// useInt returns an int slice with l elements, using i if it
+// has the necessary capacity, otherwise creating a new slice.
+func useInt(i []int, l int) []int {
```

Comment:
nit: I wouldn't use `i` and `l` as look alike for people with bad eye sight (or an aging one, like me)
perhaps replace `l` with `n` ?